src/Rendering_Internals.md

```diff
@@ -0,0 +1,408 @@
+
+Since the Pixel Slice Fetcher is shared by both FIFOs, when both of them need to be refilled at the same time, pixels temporarily stop being output until both have been served.
+
+The Pixel Slice Fetcher is told which of the 384 tiles to fetch one slice from, as well as which slice of that tile's 8, by the "fetchers"—again, one for the background and window, another for OBJs.
+These "fetchers" also directly transmit some metadata to the FIFOs[^real_fifo_refilling], such as the palette, priority, etc.
+
+[^real_fifos]:
+Actually, there are more than 2 FIFOs.
+For example, on DMG, there are [two FIFOs for BG pixel indices](https://raw.githubusercontent.com/furrtek/DMG-CPU-Inside/master/Schematics/32_BG_PIXEL_SHIFTER.png), [two for OBJ pixels](https://raw.githubusercontent.com/furrtek/DMG-CPU-Inside/master/Schematics/33_SPRITE_PIXEL_SHIFTER.png), [one for OBJ palette bits](https://raw.githubusercontent.com/furrtek/DMG-CPU-Inside/master/Schematics/34_SPRITE_PALETTE_SHIFTER.png), and [one for OBJ-to-BG priority bits](https://raw.githubusercontent.com/furrtek/DMG-CPU-Inside/master/Schematics/26_BACKGROUND.png).
+However, since many are clocked and refilled together, such as the first two or the latter four, it's easier to treat them as a single FIFO that groups all that info under one "pixel".
+
+[^real_fifo_refilling]:
+Again, since the two conceptual FIFOs are really a collection of a bunch of hardware FIFOs, what actually happens is that the "fetchers" directly refill some of these FIFOs, and instruct the Pixel Slice Fetcher to refill the corresponding pixel *data* FIFO.
+
+## The FIFOs
+
+Each FIFO can hold up to 8 pixels, the width of one tile.
+The BG FIFO and Pixel Slice Fetcher work together to ensure that the former never runs out of pixels; the OBJ FIFO is only refilled when an OBJ is "hit".
+
+Each pixel in a FIFO is composed of four properties:
+- Color index: a value between 0 and 3.
+- Palette:
+  - **On DMG**, this contains the palette bit from [OAM attributes](<#Byte 3 — Attributes/Flags>). Of course, only the OBJ FIFO has this.
+  - **On CGB**, a value between 0 and 7 (for BG, the palette bits from [BG attributes](<#BG Map attributes (CGB Mode only)>); for OBJ, the palette bits from [OAM attributes](<#Byte 3 — Attributes/Flags>)).
+- Source OBJ: only applies to the OBJ FIFO on CGB. This contains the ID of the OBJ the pixel originated from.
+- Priority:
+  - OBJ FIFO: holds the OBJ-to-BG Priority bit from [the OBJ's attributes](<#Byte 3 — Attributes/Flags>).
+  - BG FIFO, **on CGB only**: holds the OBJ-to-BG priority bit from [the tile's attributes](<#BG Map attributes (CGB Mode only)>).
+
+Every scanline, the following occurs in order:
+1. *Mode 2*:
+   1.  OAM is scanned for Y positions in range (based on [`LY`](<#FF44 — LY: LCD Y coordinate \[read-only\]>) and [`LCDC`](<#FF40 — LCDC: LCD control>)); the X coordinate is not checked!
+   2. The first 10 matches get their X and Y coordinates stored in an "OBJ slot".
```

Comment:
Isn't this different depending on whether the software is running on a DMG (including CGB with OPRI set) or a CGB?